app/services/reports/facility_progress_service.rb

```diff
@@ -122,51 +131,89 @@ def dimension_combinations_for_v1(indicator)
       end
     end
 
+    memoize def monthly_total_follow_ups
```

Comment:
Do we use memoize frequently? Since past data is allowed to be modified & syncs happen after a user is online, the cache data might be stale. Can we switch from memoize to a cache which can be invalidated?

Reply:
We use it frequently. memoize caches result for a req thread. 

We have a mechanism to invalidate the Redis cache. We invalidate the cache by passing a `force_bust`(boolean) param with the request.

Reply:
Aha, so memoize caches results inside redis? I was under the assumption it's in-memory caching like Clojure

Reply:
This comes from https://github.com/tycooon/memery. It's just caching with instance variables with some sugar. Since each request to the controller creates a new instance of `FacilityProgressService` it ends acting as a per request in-memory cache.